### PR TITLE
Categories bug fixed and tests added

### PR DESCRIPTION
**Pull request by kdefliese**
Fixed bug preventing categories page from loading
Added specs for new category page
Added cucumber tests for new and edit category actions

### CODE REVIEW

**Comment by kariabancroft** [comment]
Files in the DB folder should be deleted from PR and added to gitignore

File: features/create_category.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  I want to be able to create Categories
+
+  Background:
+    Given the blog is set up
+    And I am logged into the admin panel
+
+  Scenario: Successfully create new Categories
+    When I click on Categories
+    Then I should be on the new category page
+    When I fill in "category_name" with "Meow Meow"
```
Comment (kariabancroft) [inline features/create_category.feature:12]: What about the other fields available? Do you want to test any of those?

File: features/edit_category.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Feature: Edit Categories
+  As a blog administrator
+  I want to be able to edit Categories
+
+  Background:
+    Given the blog is set up
+    And I am logged into the admin panel
+    And I have an existing Category named "Meow"
+
+  Scenario: Successfully edit existing Categories
```
Comment (kariabancroft) [inline features/edit_category.feature:10]: Do you want to test any non-success cases?

File: spec/controllers/admin/post_types_controller_spec.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
     request.session = { :user => @user.id }
   end
 
-  it "index shoudld redirect to new" do
+  it "index should redirect to new" do
```
Comment (kariabancroft) [inline spec/controllers/admin/post_types_controller_spec.rb:13]: +1 for fixing existing code!

**Comment by kdefliese** [comment]
Updated:
- removed db files and added to gitignore
- added lines to test scenarios for other category fields
- added cucumber scenarios for creating and editing categories with nil names
- fixed bug that was causing the page to show a rails error instead of a flash error message when doing the above

**Comment by kariabancroft** [comment]
Looks good to close!

**Pull request closed without merging**